### PR TITLE
PMM-13428 Fix add remote instance issues.

### PR DESCRIPTION
**Pull request by BupycHuk**
**What this PR does / why we need it**:

It doesn't override address when node or agent is changed.
dropped node/agent selection for HAProxy and External exporter as it doesn't have effect.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.tsx [hunk 1]
```diff
@@ -39,10 +39,13 @@ export const NodesAgents: FC<NodesAgentsProps> = ({ form }) => {
   }, []);
 
   const changeAgentValue = (value: AgentsOption) => {
-    if (value.label !== PMM_SERVER_NODE_AGENT_ID) {
-      form?.change('address', 'localhost');
-    } else {
-      form?.change('address', '');
+    console.log(form?.getState().values?.address);
```
Comment (ademidoff) [inline public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.tsx:42]:
```suggestion
```

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.tsx [hunk 2]
```diff
@@ -39,10 +39,13 @@ export const NodesAgents: FC<NodesAgentsProps> = ({ form }) => {
...
+    if (form?.getState().values?.address === undefined || form?.getState().values?.address === '' ) {
```
Comment (ademidoff) [inline public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.tsx:43]:
```suggestion
    if (!form?.getState().values?.address) {
```

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.types.ts [hunk 3]
```diff
@@ -2,4 +2,5 @@ import { FormApi } from 'final-form';
 
 export interface NodesAgentsProps {
   form?: FormApi;
+  isRDS?: boolean;
```
Comment (matejkubinec) [inline public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.types.ts:5]: This appears not to be used.

**Review by matejkubinec — COMMENTED** [review]
Outside of the one comment, just what Alex pointed out.

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.types.ts (hunk 3 above)
Comment (BupycHuk) [reply public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.types.ts:5]: yes, it's leftover, we can drop it

**Review by matejkubinec — APPROVED** [review]

**Review by YashSartanpara1 — APPROVED** [review]

**Pull request merged**